testutil/integration/suites/update_params.go

```diff
@@ -0,0 +1,424 @@
+			NewParamClientFn:    prooftypes.NewQueryClient,
+		},
+		tokenomicstypes.ModuleName: {
+			ValidParams:             tokenomicstypes.Params{},
+			MsgUpdateParams:         tokenomicstypes.MsgUpdateParams{},
+			MsgUpdateParamsResponse: tokenomicstypes.MsgUpdateParamsResponse{},
+			QueryParamsRequest:      tokenomicstypes.QueryParamsRequest{},
+			QueryParamsResponse:     tokenomicstypes.QueryParamsResponse{},
+			DefaultParams:           tokenomicstypes.DefaultParams(),
+			NewParamClientFn:        tokenomicstypes.NewQueryClient,
+		},
+	}
+
+	// MsgUpdateParamEnabledModuleNames is a list of module names which support
+	// individual parameter updates (i.e. MsgUpdateParam). It is initialized in
+	// init().
+	MsgUpdateParamEnabledModuleNames []string
+
+	_ IntegrationSuite = (*UpdateParamsSuite)(nil)
+)
+
+func init() {
+	for moduleName, moduleThing := range ModuleParamConfigMap {
+		if moduleThing.MsgUpdateParam != nil {
+			MsgUpdateParamEnabledModuleNames = append(MsgUpdateParamEnabledModuleNames, moduleName)
+		}
+	}
+}
+
+// UpdateParamsSuite is an integration test suite that provides helper functions for
+// running parameter update messages. It is intended to be embedded in other integration
+// test suites which are dependent on parameter updates.
+type UpdateParamsSuite struct {
+	AuthzIntegrationSuite
+}
+
+// SetupTestAuthzAccounts sets AuthorityAddr for the suite by getting the authority
+// from the integration app. It also assigns a new pre-generated identity to be used
+// as the AuthorizedAddr for the suite. It is expected to be called after s.NewApp()
+// as it depends on the integration app and its pre-generated account iterator.
+func (s *UpdateParamsSuite) SetupTestAuthzAccounts() {
+	// Set the authority, authorized, and unauthorized addresses.
+	AuthorityAddr = cosmostypes.MustAccAddressFromBech32(s.GetApp().GetAuthority())
+
+	nextAcct, ok := s.GetApp().GetPreGeneratedAccounts().Next()
+	require.True(s.T(), ok, "insufficient pre-generated accounts available")
+	AuthorizedAddr = nextAcct.Address
+}
+
+// SetupTestAuthzGrants creates on-chain authz grants for the MsgUpdateUpdateParam and
+// MsgUpdateParams message for each module. It is expected to be called after s.NewApp()
+// as it depends on the authority and authorized addresses having been set.
+func (s *UpdateParamsSuite) SetupTestAuthzGrants() {
+	// Create authz grants for all poktroll modules' MsgUpdateParams messages.
+	s.RunAuthzGrantMsgForPoktrollModules(s.T(),
+		AuthorityAddr,
+		AuthorizedAddr,
+		MsgUpdateParamsName,
+		s.GetPoktrollModuleNames()...,
+	)
+
+	// Create authz grants for all poktroll modules' MsgUpdateParam messages.
+	s.RunAuthzGrantMsgForPoktrollModules(s.T(),
+		AuthorityAddr,
+		AuthorizedAddr,
+		MsgUpdateParamName,
+		// NB: only modules with params are expected to support MsgUpdateParam.
+		MsgUpdateParamEnabledModuleNames...,
+	)
+}
+
+// RunUpdateParams runs the given MsgUpdateParams message via an authz exec as the
```

Comment:
Any chance you can put together a small readme (or just like 3 slides) on:
- authority addr
- authz exec
- what this enables

I'll find it super useful and I believe the rest of the team would too.

Reply:
I've added it to my documentation todo list. :+1:

Reply:
Can you just add TODO_DOCUMENT in docasurus as well? It's why if the page is empty. At least, we'll know what you plan to document (it helps me too)